fs2-google-pubsub-http/src/main/scala/com/permutive/pubsub/consumer/http/internal/PubsubSubscriber.scala

```diff
@@ -60,11 +62,14 @@ private[http] object PubsubSubscriber {
         )
       msg <- Stream.emits(
         rec.receivedMessages.map { msg =>
-          Model.Record(
-            value = msg.message,
-            ack = ackQ.enqueue1(msg.ackId),
-            nack = nackQ.enqueue1(msg.ackId)
-          )
+          new InternalRecord[F] {
+            override val value: PubsubMessage = msg.message
+            override val ack: F[Unit]         = ackQ.enqueue1(msg.ackId)
+            override val nack: F[Unit]        = nackQ.enqueue1(msg.ackId)
+            override def extendDeadline(by: FiniteDuration): F[Unit] =
+              reader.modifyDeadline(List(msg.ackId), by)
```

Comment:
I think I can see _why_ you'd make this synchronous (extensions has to happen quite soon after it's requested though), might we want to add a new `extension` queue that can be configured separately to the ack one though, just for possible performance? Default settings could have a lower `maxDelay`?

Reply:
I was thinking of doing that, but was worried that it will happen too late. I feel like extending deadlines should be a relatively uncommon thing so extending synchronously should be ok..

Reply:
Seems fair for a first go-around :+1: